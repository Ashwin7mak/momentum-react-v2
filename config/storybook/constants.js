module.exports = {
  REPO_ROOT: '../../../',
  SRC_DIR: './src/',
  ESM_DIR: './dist/esm/',
  CJS_DIR: './dist/cjs/',
  COMPONENTS_DIR: './components/',
};
