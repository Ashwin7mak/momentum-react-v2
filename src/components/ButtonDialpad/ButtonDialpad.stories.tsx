import React, { FC } from 'react';
import { Story } from '@storybook/react';
import {
  Title,
  Subtitle,
  Description,
  Primary,
  ArgsTable,
  PRIMARY_STORY,
} from '@storybook/addon-docs';

import ButtonDialPad, { ButtonDialpadProps, BUTTON_DIALPAD_CONSTANTS as CONSTANTS } from './';
import Documentation from './ButtonDialpad.documentation.mdx';

const DocsPage: FC = () => (
  <>
    <Title />
    <Subtitle />
    <Description />
    <Documentation />
    <Primary />
    <ArgsTable story={PRIMARY_STORY} />
  </>
);

export default {
  title: 'Momentum UI/ButtonDialpad',
  component: ButtonDialPad,
  parameters: {
    expanded: true,
    docs: {
      page: DocsPage,
    },
  },
};

const argTypes = {
  children: {
    defaultValue: '',
    description: 'Provides the child node for this element.',
    control: { type: 'text' },
    table: {
      type: {
        summary: 'ReactNode',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  disabled: {
    defaultValue: CONSTANTS.DEFAULTS.DISABLED,
    description: 'Whether to render the `<ButtonDialpad />` is disabled.',
    options: [true, false],
    control: { type: 'boolean' },
    table: {
      type: {
        summary: 'boolean',
      },
      defaultValue: {
        summary: CONSTANTS.DEFAULTS.DISABLED,
      },
    },
  },
  primaryText: {
    defaultValue: '1',
    description: 'Provides the primary text for this `<ButtonDialPad />`.',
    control: { type: 'text' },
    table: {
      type: {
        summary: 'string',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  secondaryText: {
    defaultValue: 'ABC',
    description: 'Provides the prsecondaryimary text for this `<ButtonDialPad />`.',
    control: { type: 'text' },
    table: {
      type: {
        summary: 'string',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  size: {
    defaultValue: CONSTANTS.DEFAULTS.SIZE,
    description: 'Modifies the size of this `<ButtonPill />`.',
    options: [undefined, ...Object.values(CONSTANTS.SIZES)],
    control: { type: 'select' },
    table: {
      type: {
        summary: 'number',
      },
      defaultValue: {
        summary: CONSTANTS.DEFAULTS.SIZE,
      },
    },
  },
  onPress: {
    action: 'onPress',
    description:
      'From [AriaButtonProps](https://react-spectrum.adobe.com/react-aria/useButton.html). Handler that is called when the press is released over the target.',
    table: {
      category: 'AriaButtonProps',
      type: {
        summary: '(e: PressEvent) => void',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  autoFocus: {
    action: 'autoFocus',
    description:
      'From [AriaButtonProps](https://react-spectrum.adobe.com/react-aria/useButton.html). Whether the element should receive focus on render.',
    table: {
      category: 'AriaButtonProps',
      type: {
        summary: 'boolean',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
  onKeyDown: {
    action: 'onKeyDown',
    description:
      'From [AriaButtonProps](https://react-spectrum.adobe.com/react-aria/useButton.html). Handler that is called when a key is pressed.',
    table: {
      category: 'AriaButtonProps',
      type: {
        summary: '(e: KeyboardEvent) => void',
      },
      defaultValue: {
        summary: 'undefined',
      },
    },
  },
};

const Template: Story<ButtonDialpadProps> = (args: ButtonDialpadProps) => {
  const mutatedArgs = { ...args };
  delete mutatedArgs.children;

  return <ButtonDialPad {...mutatedArgs}>{args.children}</ButtonDialPad>;
};

const MultiTemplate: Story<ButtonDialpadProps> = (args: ButtonDialpadProps, { parameters }) => {
  const mutatedArgs = { ...args };
  const { children } = mutatedArgs;
  delete mutatedArgs.children;

  const { variants } = parameters;

  const items = variants.map((variant, index: number) => (
    <ButtonDialPad key={index} {...variant} {...args}>
      {children}
    </ButtonDialPad>
  ));

  return <>{items}</>;
};

const Example = Template.bind({});

Example.argTypes = { ...argTypes };

const Sizes = MultiTemplate.bind({});

Sizes.parameters = {
  variants: [{}, { size: 52 }],
};

Sizes.argTypes = { ...argTypes };
delete Sizes.argTypes.size;

const Common = MultiTemplate.bind({});

const States = MultiTemplate.bind({});

States.parameters = {
  variants: [{}, { disabled: true }],
};

States.argTypes = { ...argTypes };
delete States.argTypes.disabled;

Common.parameters = {
  variants: [
    { primaryText: '1' },
    { primaryText: '2', secondaryText: 'ABC' },
    { primaryText: '3', secondaryText: 'DEF' },
    { primaryText: '4', secondaryText: 'GHI' },
    { primaryText: '5', secondaryText: 'JKL' },
    { primaryText: '6', secondaryText: 'MNO' },
    { primaryText: '7', secondaryText: 'PQRS' },
    { primaryText: '8', secondaryText: 'TUV' },
    { primaryText: '9', secondaryText: 'WXYZ' },
    { primaryText: '0' },
    { primaryText: '*' },
    { primaryText: '#' },
  ],
};

Common.argTypes = { ...argTypes };
delete Common.argTypes.children;
delete Common.argTypes.primaryText;
delete Common.argTypes.secondaryText;

export { Example, Sizes, States, Common };
