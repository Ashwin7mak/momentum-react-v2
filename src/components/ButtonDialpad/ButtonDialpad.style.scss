.md-button-dialpad-wrapper {
  border: var(--border-clear);
  cursor: pointer;
  font-family: var(--font-default);
  margin: 0.5rem;
  outline: none;
  transition: background-color 0.2s, color 0.2s, border-color 0.2s;
  background-color: var(--button-dialpad-background);

  &:hover {
    background-color: var(--button-dialpad-background-hovered);
  }

  &:active {
    background-color: var(--button-dialpad-background-pressed);
  }

  > * {
    align-items: center;
    display: flex;
    height: 1em;
    justify-content: center;
  }

  > .md-button-dialpad-primary-text {
    color: var(--button-dialpad-special-text-number);
  }

  > .md-button-dialpad-secondary-text {
    color: var(--button-dialpad-special-text-characters);
    margin-top: 0.2em;
  }

  &[data-disabled='true'] {
    background-color: var(--button-dialpad-background-disabled);
    color: var(--button-dialpad-text-number);
    cursor: auto;

    > .md-button-dialpad-primary-text {
      color: var(--button-dialpad-text-number);
    }

    > .md-button-dialpad-secondary-text {
      color: var(--button-dialpad-text-characters);
    }

    &:hover {
      background-color: var(--button-dialpad-background-disabled);
    }

    &:active {
      background-color: var(--button-dialpad-background-disabled);
    }
  }

  /* Sizing and spacing. */
  align-items: center;
  border-radius: 100vh;
  display: flex;
  flex-direction: column;
  flex-shrink: 0;
  justify-content: center;

  &[data-size='64'] {
    height: 4rem;
    width: 4rem;

    > .md-button-dialpad-primary-text {
      font-size: 1.625rem;
    }

    > .md-button-dialpad-secondary-text {
      font-size: 0.75rem;
    }
  }

  &[data-size='52'] {
    height: 3.25rem;
    width: 3.25rem;

    > .md-button-dialpad-primary-text {
      font-size: 1.25rem;
    }

    > .md-button-dialpad-secondary-text {
      font-size: 0.75rem;
    }
  }
}
