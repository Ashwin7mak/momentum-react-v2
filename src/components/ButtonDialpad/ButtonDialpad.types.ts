import { ReactNode } from 'react';
import { AriaButtonProps } from '@react-types/button';

export interface Props extends AriaButtonProps {
  /**
   * Custom class to be able to override the component's CSS
   */
  className?: string;
  /**
   * Child components of this ButtonDialpad. This prop replaces primaryText.
   */
  children?: ReactNode;
  /**
   * Whether or not this ButtonDialpad is disabled.
   */
  disabled?: boolean;

  /**
   * Primary text to display above the secondary text.
   */
  primaryText?: string;

  /**
   * Secondary text to display below the primary text.
   */
  secondaryText?: string;

  /**
   * Size index of this ButtonPill.
   */
  size?: number;

  /**
   * title to use for this component.
   */
  title?: string;
}
