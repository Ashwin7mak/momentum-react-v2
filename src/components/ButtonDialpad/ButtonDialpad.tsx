import React, { forwardRef, useRef, FC, RefObject } from 'react';
import { useButton } from '@react-aria/button';
import FocusRing from '../FocusRing';

import { DEFAULTS, STYLE } from './ButtonDialpad.constants';
import { Props } from './ButtonDialpad.types';
import './ButtonDialpad.style.scss';
import classnames from 'classnames';

const ButtonDialpad: FC<Props> = forwardRef(
  (props: Props, providedRef: RefObject<HTMLButtonElement>) => {
    const { className } = props;
    const ref = providedRef || useRef();
    const mutatedProps = {
      ...props,
      isDisabled: props.disabled,
    };

    delete mutatedProps.disabled;
    delete mutatedProps.className;

    const { buttonProps } = useButton(mutatedProps, ref);
    const children = props.children || props.primaryText;

    return (
      <FocusRing disabled={props.disabled}>
        <button
          className={classnames(STYLE.wrapper, className)}
          {...buttonProps}
          ref={ref}
          data-size={props.size || DEFAULTS.SIZE}
          data-disabled={props.disabled || DEFAULTS.DISABLED}
        >
          <div className={STYLE.primaryText}>{children}</div>
          <div className={STYLE.secondaryText}>{props.secondaryText}</div>
        </button>
      </FocusRing>
    );
  }
);

ButtonDialpad.displayName = 'ButtonDialpad';

export default ButtonDialpad;
