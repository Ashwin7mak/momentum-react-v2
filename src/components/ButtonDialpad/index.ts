import { default as ButtonDialpad } from './ButtonDialpad';
import * as CONSTANTS from './ButtonDialpad.constants';
import { Props } from './ButtonDialpad.types';

export { CONSTANTS as BUTTON_DIALPAD_CONSTANTS };

export type ButtonDialpadProps = Props;

export default ButtonDialpad;
