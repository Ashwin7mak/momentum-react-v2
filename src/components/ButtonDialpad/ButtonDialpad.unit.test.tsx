import React from 'react';
import { mount } from 'enzyme';
import { ButtonDialpad } from '@momentum-ui/react';

import { DEFAULTS, SIZES, STYLE } from './ButtonDialpad.constants';


describe('<ButtonPill />', () => {
  let container;

  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      container = mount(<ButtonDialpad>1</ButtonDialpad>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with primaryText', () => {
      expect.assertions(1);

      container = mount(<ButtonDialpad primaryText="1" />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with secondaryText', () => {
      expect.assertions(1);

      container = mount(<ButtonDialpad secondaryText="ABC" />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with primaryText and secondaryText', () => {
      expect.assertions(1);

      container = mount(<ButtonDialpad primaryText="1" secondaryText="ABC" />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with primaryText and children', () => {
      expect.assertions(1);

      container = mount(<ButtonDialpad primaryText="1">2</ButtonDialpad>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with primaryText, secondaryText and children', () => {
      expect.assertions(1);

      container = mount(<ButtonDialpad primaryText="1" secondaryText="ABC">2</ButtonDialpad>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with size', () => {
      expect.assertions(1);

      const size = SIZES[Object.keys(SIZES)[Object.keys(SIZES).length - 1]];

      container = mount(<ButtonDialpad size={size}>Example Text</ButtonDialpad>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when disabled', () => {
      expect.assertions(1);

      const disabled = !DEFAULTS.DISABLED;

      container = mount(<ButtonDialpad disabled={disabled}>Example Text</ButtonDialpad>);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its main class', () => {
      expect.assertions(1);

      const element = mount(<ButtonDialpad />).find(ButtonDialpad).getDOMNode();

      expect(element.classList.contains(STYLE.wrapper)).toBe(true);
    });

    it('should pass disabled prop', () => {
      expect.assertions(1);

      const disabled = !DEFAULTS.DISABLED;

      const element = mount(<ButtonDialpad disabled={disabled} />).find(ButtonDialpad).getDOMNode();

      expect(element.getAttribute('data-disabled')).toBe(`${disabled}`);
    });

    it('should pass size prop', () => {
      expect.assertions(1);

      const size = SIZES[Object.keys(SIZES)[Object.keys(SIZES).length - 1]];

      const element = mount(<ButtonDialpad size={size} />).find(ButtonDialpad).getDOMNode();

      expect(element.getAttribute('data-size')).toBe(`${size}`);
    });

    it('should pass child prop', () => {
      expect.assertions(1);

      const child = '1';

      const element = mount(<ButtonDialpad>{child}</ButtonDialpad>).find(ButtonDialpad).getDOMNode();
      const target = element.getElementsByClassName(STYLE.primaryText)[0];

      expect(target.innerHTML).toBe(child);
    });

    it('should pass primaryText prop', () => {
      expect.assertions(1);

      const primaryText = '1';

      const element = mount(<ButtonDialpad primaryText={primaryText} />).find(ButtonDialpad).getDOMNode();
      const target = element.getElementsByClassName(STYLE.primaryText)[0];

      expect(target.innerHTML).toBe(primaryText);
    });

    it('should pass secondaryText prop', () => {
      expect.assertions(1);

      const secondaryText = 'ABC';

      const element = mount(<ButtonDialpad secondaryText={secondaryText} />).find(ButtonDialpad).getDOMNode();
      const target = element.getElementsByClassName(STYLE.secondaryText)[0];

      expect(target.innerHTML).toBe(secondaryText);
    });

    it('should override primaryText prop with child prop', () => {
      expect.assertions(2);

      const primaryText = '1';
      const child = '2';

      const element = mount(<ButtonDialpad primaryText={primaryText}>{child}</ButtonDialpad>).find(ButtonDialpad).getDOMNode();
      const target = element.getElementsByClassName(STYLE.primaryText)[0];

      expect(target.innerHTML).toBe(child);
      expect(target.innerHTML).not.toBe(primaryText);
    });
  });

  describe('actions', () => {
    it('should handle mouse press events', () => {
      expect.assertions(1);

      const mockCallback = jest.fn();

      const component = mount(<ButtonDialpad onPress={mockCallback} />).find(ButtonDialpad);

      component.props().onPress({
        type: 'press',
        pointerType: 'mouse',
        shiftKey: false,
        ctrlKey: false,
        metaKey: false,
        target: component.getDOMNode(),
      });

      expect(mockCallback).toBeCalledTimes(1);
    });
  });
});
