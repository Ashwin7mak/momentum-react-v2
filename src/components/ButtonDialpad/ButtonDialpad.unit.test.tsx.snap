// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonPill /> snapshot should match snapshot 1`] = `
<ButtonDialpad>
  <FocusRing
    focusRingClass="md-button-dialpad-focus-ring"
  >
    <button
      className="md-button-dialpad-wrapper"
      data-disabled={false}
      data-size={64}
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="md-button-dialpad-primary-text"
      >
        1
      </div>
      <div
        className="md-button-dialpad-secondary-text"
      />
    </button>
  </FocusRing>
</ButtonDialpad>
`;

exports[`<ButtonPill /> snapshot should match snapshot when disabled 1`] = `
<ButtonDialpad
  disabled={true}
>
  <FocusRing
    focusRingClass="md-button-dialpad-focus-ring"
  >
    <button
      className="md-button-dialpad-wrapper"
      data-disabled={true}
      data-size={64}
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="md-button-dialpad-primary-text"
      >
        Example Text
      </div>
      <div
        className="md-button-dialpad-secondary-text"
      />
    </button>
  </FocusRing>
</ButtonDialpad>
`;

exports[`<ButtonPill /> snapshot should match snapshot with primaryText 1`] = `
<ButtonDialpad
  primaryText="1"
>
  <FocusRing
    focusRingClass="md-button-dialpad-focus-ring"
  >
    <button
      className="md-button-dialpad-wrapper"
      data-disabled={false}
      data-size={64}
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="md-button-dialpad-primary-text"
      >
        1
      </div>
      <div
        className="md-button-dialpad-secondary-text"
      />
    </button>
  </FocusRing>
</ButtonDialpad>
`;

exports[`<ButtonPill /> snapshot should match snapshot with primaryText and children 1`] = `
<ButtonDialpad
  primaryText="1"
>
  <FocusRing
    focusRingClass="md-button-dialpad-focus-ring"
  >
    <button
      className="md-button-dialpad-wrapper"
      data-disabled={false}
      data-size={64}
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="md-button-dialpad-primary-text"
      >
        2
      </div>
      <div
        className="md-button-dialpad-secondary-text"
      />
    </button>
  </FocusRing>
</ButtonDialpad>
`;

exports[`<ButtonPill /> snapshot should match snapshot with primaryText and secondaryText 1`] = `
<ButtonDialpad
  primaryText="1"
  secondaryText="ABC"
>
  <FocusRing
    focusRingClass="md-button-dialpad-focus-ring"
  >
    <button
      className="md-button-dialpad-wrapper"
      data-disabled={false}
      data-size={64}
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="md-button-dialpad-primary-text"
      >
        1
      </div>
      <div
        className="md-button-dialpad-secondary-text"
      >
        ABC
      </div>
    </button>
  </FocusRing>
</ButtonDialpad>
`;

exports[`<ButtonPill /> snapshot should match snapshot with primaryText, secondaryText and children 1`] = `
<ButtonDialpad
  primaryText="1"
  secondaryText="ABC"
>
  <FocusRing
    focusRingClass="md-button-dialpad-focus-ring"
  >
    <button
      className="md-button-dialpad-wrapper"
      data-disabled={false}
      data-size={64}
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="md-button-dialpad-primary-text"
      >
        2
      </div>
      <div
        className="md-button-dialpad-secondary-text"
      >
        ABC
      </div>
    </button>
  </FocusRing>
</ButtonDialpad>
`;

exports[`<ButtonPill /> snapshot should match snapshot with secondaryText 1`] = `
<ButtonDialpad
  secondaryText="ABC"
>
  <FocusRing
    focusRingClass="md-button-dialpad-focus-ring"
  >
    <button
      className="md-button-dialpad-wrapper"
      data-disabled={false}
      data-size={64}
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="md-button-dialpad-primary-text"
      />
      <div
        className="md-button-dialpad-secondary-text"
      >
        ABC
      </div>
    </button>
  </FocusRing>
</ButtonDialpad>
`;

exports[`<ButtonPill /> snapshot should match snapshot with size 1`] = `
<ButtonDialpad
  size={64}
>
  <FocusRing
    focusRingClass="md-button-dialpad-focus-ring"
  >
    <button
      className="md-button-dialpad-wrapper"
      data-disabled={false}
      data-size={64}
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="md-button-dialpad-primary-text"
      >
        Example Text
      </div>
      <div
        className="md-button-dialpad-secondary-text"
      />
    </button>
  </FocusRing>
</ButtonDialpad>
`;
